### PR TITLE
BB-987: Add variables needed for setting up the User Retirement pipeline

### PR DESCRIPTION
**Pull request by giovannicimolin**
This PR adds the required variables to correctly set up the User Retirement pipeline inside our AppServers.

**Testing instructions:**
1. Pull this branch on your local Ocim devstack,
2. Start your devstack and SSH into it (`vagrant up && vagrant ssh`)
3. Enter django shell (`make shell`) and spawn a new OpenEdXInstance using the following settings:
```
instance = OpenEdXInstance.objects.create(
    name='bb-987-test',
    sub_domain="bb-987-test",
    openedx_release="master",
    configuration_source_repo_url="https://github.com/open-craft/configuration.git",
    configuration_version=giovanni/bb-987,
)
```
4. Comment out or put fake OVH credentials on the `.env` file (we don't want to spawn any instance at this point).
5. Spawn a new AppServer using Ocim we interface (you might need to hit the button twice due to the load balancer configuration failing)
6. When the appserver shows up on the list, check that the correct Ansible variables are set:
```
EDXAPP_RETIREMENT_STATES:
- PENDING
- RETIRING_ENROLLMENTS
- ENROLLMENTS_COMPLETE
- RETIRING_LMS_MISC
- LMS_MISC_COMPLETE
- RETIRING_LMS
- LMS_COMPLETE
- RETIRING_CREDENTIALS
- CREDENTIALS_COMPLETE
- ERRORED
- ABORTED
- COMPLETE
...
RETIREMENT_CREDENTIALS_BASE_BASE_URL: http://localhost:8150
RETIREMENT_ECOMMERCE_BASE_BASE_URL: https://ecommerce.hawthorn-giovanni-test.plebia.net
RETIREMENT_LMS_BASE_URL: https://hawthorn-giovanni-test.plebia.net
RETIREMENT_SERVICE_ENABLE_CRON_JOB: true
RETIREMENT_SERVICE_OAUTH_CLIENT_ID: SOMETHING-HERE
RETIREMENT_SERVICE_OAUTH_CLIENT_SECRET: SOMETHING-ELSE-HERE
RETIREMENT_SERVICE_PIPELINE_CONFIGURATION:
- FUNCTION: retirement_unenroll
  NAME: RETIRING_ENROLLMENTS
  NAME_COMPLETE: ENROLLMENTS_COMPLETE
  SERVICE: LMS
- FUNCTION: retirement_lms_retire_misc
  NAME: RETIRING_LMS_MISC
  NAME_COMPLETE: LMS_MISC_COMPLETE
  SERVICE: LMS
- FUNCTION: retirement_lms_retire
  NAME: RETIRING_LMS
  NAME_COMPLETE: LMS_COMPLETE
  SERVICE: LMS
RETIREMENT_SERVICE_SETUP: true
...
"EDXAPP_FEATURES": {
    ...
    "ENABLE_ACCOUNT_DELETION": true
}
```

**Deployment checking:**
I have already tested the deployment of the User Retirement pipeline on [this Ocim instance.](https://stage.console.opencraft.com/instance/3360/)
1. SSH into https://stage.console.opencraft.com/instance/3360/.
2. Log in as root user with `sudo su root -s /bin/bash`
3. Check if the User Retirement script was correctly set up on `/edx/app/retirement_service`
4. Check that the cron job is scheduled to run this task daily at midnight.
5. Create a user and mark it's account for deletion.
6. Run the retirement script as root:
```
/edx/app/retirement_service/retire_users.sh
```
You might find useful to edit `/edx/app/retirement_service/retire_users.sh` and change `--cool_off_days=5` to `--cool_off_days=0` instead of waiting 5 days to check if the user was deleted. :stuck_out_tongue:

**Reviewers:**
- [x] @jamestait

### CODE REVIEW

File: instance/models/mixins/openedx_config.py [hunk 1]
```diff
@@ -210,6 +210,7 @@ def _get_configuration_variables(self):
                 "REQUIRE_COURSE_EMAIL_AUTH": False,
                 "USE_MICROSITES": False,
                 "PREVENT_CONCURRENT_LOGINS": False,
+                "ENABLE_ACCOUNT_DELETION": True,
```
[the remark this answers is not in the document]
Comment (giovannicimolin) [reply instance/models/mixins/openedx_config.py]: Nope, this is only the feature flag that enables the "Delete my account" section on LMS User account.

File: instance/models/mixins/openedx_config.py [hunk 2]
```diff
@@ -310,6 +311,50 @@ def _get_configuration_variables(self):
             # in Ocim deployments.
             # Disabling heartbeats can have a drastic reduction RabbitMQ usage.
             "worker_django_enable_heartbeats": settings.EDX_WORKERS_ENABLE_CELERY_HEARTBEATS,
+
+            # Set up User Retirement Pipeline
+            "RETIREMENT_SERVICE_SETUP": True,
+            "RETIREMENT_SERVICE_ENABLE_CRON_JOB": True,
+            "RETIREMENT_SERVICE_COOL_OFF_DAYS": 5,
```
[the remark this answers is not in the document]
Comment (giovannicimolin) [reply instance/models/mixins/openedx_config.py]: Right, I'll add the cron job scheduling variables too.

[the remark this answers is not in the document]
Comment (giovannicimolin) [reply instance/models/mixins/openedx_config.py]: @jamestait Done.

**Comment by jamestait** [comment]
I had to disable SSL certificate verification to get this to work, as the Stage server uses certificates issued by an invalid CA.

```root@edxapp-bb-987stageop-appserver-4:/edx/app/retirement_service# /edx/app/retirement_service/retire_users.sh
Writing logs to: [ /edx/var/log/retirement_service ]     
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): bb-987.stage.opencraft.hosting:443                                                                            
DEBUG:urllib3.connectionpool:https://bb-987.stage.opencraft.hosting:443 "POST /oauth2/access_token HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): bb-987.stage.opencraft.hosting:443
DEBUG:urllib3.connectionpool:https://bb-987.stage.opencraft.hosting:443 "GET /api/user/v1/accounts/retirement_queue/?cool_off_days=0&states=PENDING&states=ENROLLMENTS_COMPLETE&states=LMS_MISC_COMPLETE&states=LMS_COMPLETE HTTP/1.1" 200 None
/edx/app/retirement_service/venv/lib/python3.5/site-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.ht
ml#ssl-warnings                                              
  InsecureRequestWarning)                     
...
b'Learner Retirement: Starting learner retirement for jamestait using config file /edx/app/retirement_service/config.yml'
b'Learner Retirement: Starting state RETIRING_ENROLLMENTS'
b'Learner Retirement: State RETIRING_ENROLLMENTS completed in 0.11644172668457031 seconds'
b'Learner Retirement: Progressing to state ENROLLMENTS_COMPLETE'
b'Learner Retirement: Starting state RETIRING_LMS_MISC'
b'Learner Retirement: State RETIRING_LMS_MISC completed in 0.1486492156982422 seconds'
b'Learner Retirement: Progressing to state LMS_MISC_COMPLETE'
b'Learner Retirement: Starting state RETIRING_LMS'
b'Learner Retirement: State RETIRING_LMS completed in 1.370851755142212 seconds'
b'Learner Retirement: Progressing to state LMS_COMPLETE'
b'Learner Retirement: Retirement complete for learner jamestait'
```

Verified after this that I'm unable to login.

**Pull request merged**